src/pint/bayesian.py

```diff
@@ -0,0 +1,184 @@
+import numpy as np
+import scipy.special
+
+import pint.models
+import pint.toa
+import pint.residuals
+
+class Prior:
+    def __init__(self, model : pint.models.TimingModel, dist : str, width=1, info=None):
+        self.params = model.free_params
+        self.dist = dist
+        if self.dist in ['uniform', 'normal']:
+            values = np.array([getattr(model,param).value for param in self.params])
+            uncertainties = np.array([getattr(model,param).uncertainty.value for param in self.params])
+            
+            if np.any(uncertainties==0):
+                raise ValueError("All fitting parameters must have non-zero uncertainties for this method to work. Update uncertainties in the par file or use dist='custom'.")
```

Comment:
Are the uncertainties meant to establish the priors? Uncertainties are normally set as the result of a fitting operation, so they are usually an estimate of the posterior.